### PR TITLE
Add SSR and SSG detection for Next.js

### PR DESCRIPTION
**Pull request by tunetheweb**
As discussed in https://github.com/HTTPArchive/wappalyzer/pull/6#issuecomment-1854092228

FYI @devknoll / @timneutkens would love your review of this.

### CODE REVIEW

File: src/technologies/n.json [hunk 1]
```diff
@@ -682,6 +682,39 @@
...
+      ".__NEXT_DATA__.gssp": "true"
+    },
+    "requires": "Next.js",
+    "website": "https://nextjs.org/docs/pages/building-your-application/rendering/server-side-rendering"
+  },
+  "Next.js SSG": {
+    "cats": [
+      12
+    ],
+    "description": "Next.js Statis Site Generation.",
```
Comment (rviscomi) [inline src/technologies/n.json:705]:
```suggestion
    "description": "Next.js Static Site Generation.",
```

File: src/technologies/n.json [hunk 2]
```diff
@@ -682,6 +682,39 @@
...
+    "js": {
+      ".__NEXT_DATA__.gip": "true",
+      ".__NEXT_DATA__.gssp": "true"
+    },
+    "requires": "Next.js",
+    "website": "https://nextjs.org/docs/pages/building-your-application/rendering/server-side-rendering"
+  },
+  "Next.js SSG": {
+    "cats": [
+      12
```
Comment (rviscomi) [inline src/technologies/n.json:703]:
```suggestion
      57
```

Comment (rviscomi) [reply src/technologies/n.json:703]: I think we should also remove 57 from Next.js above

File: src/technologies/n.json [hunk 3]
```diff
@@ -682,6 +682,39 @@
     "requires": "Next.js",
     "website": "https://nextjs.org/docs/app"
   },
+  "Next.js SSR": {
+    "cats": [
+      12
```
Comment (rviscomi) [inline src/technologies/n.json:687]:
```suggestion
      22
```

Comment (rviscomi) [reply src/technologies/n.json:687]: I think we should also remove 22 from Next.js above

File: src/technologies/n.json (hunk 2 above)
Comment (tunetheweb) [reply src/technologies/n.json]: I've done that, but based on https://github.com/HTTPArchive/wappalyzer/pull/6#issuecomment-1854628880 I've included them in App Router as they might be SSG

File: src/technologies/n.json (hunk 3 above)
Comment (tunetheweb) [reply src/technologies/n.json]: I've done that, but based on https://github.com/HTTPArchive/wappalyzer/pull/6#issuecomment-1854628880 I've included them in App Router as they might be SSG

**Review by rviscomi — APPROVED** [review]

File: src/technologies/n.json [hunk 4]
```diff
@@ -669,7 +667,9 @@
   },
   "Next.js App Router": {
     "cats": [
-      12
+      12,
+      22,
+      57
```
[the remark this answers is not in the document]
Comment (timneutkens) [reply src/technologies/n.json:672]: I'll become harder to detect when partial prerendering is fully rolled out as a page might be mostly static but then stream in dynamic parts.

[the remark this answers is not in the document]
Comment (tunetheweb) [reply src/technologies/n.json:672]: Yeah maybe we should remove this as it's only "maybe SSR". Only issue then is that we'll see "Next SSR" (for non-App router sites) but without the Next App router sites - even though some of these might be faster, but think that's the right thing to do despite that.

```suggestion
      12,
      22
```

**Comment by tunetheweb** [comment]
Will give @timneutkens a couple more days to review my reply to him and then merge this Monday evening (Irish time) so it's also ready to go for next crawl on Tuesday/Wednesday.

**Pull request merged**